lib/db/RoutingDAO.js

```diff
@@ -0,0 +1,280 @@
+import ArrayUtils from '@/lib/ArrayUtils';
+import { centrelineKey, CentrelineType } from '@/lib/Constants';
+import db from '@/lib/db/db';
+import CentrelineDAO from '@/lib/db/CentrelineDAO';
+
+/**
+ * @typedef {Object} IntersectionRoutingResult
+ * @property {number} cost - total cost of the route
+ * @property {Array<CentrelineFeature>} route - list of features representing the route
+ */
+
+/**
+ * @typedef {Object} FeatureRoutingResult
+ * @property {CentrelineFeature} next - next feature to continue routing from
+ * @property {Array<CentrelineFeature>} route - list of features representing the route
+ */
+
+/**
+ * Data access layer for centreline routing.  This was originally added to support corridor
+ * interpolation for multi-location selections.
+ *
+ * A route is a (possibly empty) sequence of alternating intersections and midblocks connecting
+ * zero or more waypoint features.  Although in general any set of two or more waypoints will
+ * have many such routes connecting them, this class always looks for the single "best" route
+ * using shortest-path algorithms available in the `pgrouting` extension.
+ *
+ * Note that these methods return routes as a sequence of {@link CentrelineFeature} objects.  If
+ * you need the entire location metadata, use {@link CentrelineDAO.byFeatures} to resolve the
+ * resulting features to locations.
+ */
+class RoutingDAO {
+  /**
+   *
+   * @param {number} intersectionFrom - intersection to route from
+   * @param {number} intersectionTo - intersection to route to
+   * @returns {IntersectionRoutingResult?} best route between `intersectionFrom` and
+   * `intersectionTo`, excluding `intersectionFrom` but including `intersectionTo`,
+   * or `null` if no such route exists
+   */
+  static async routeIntersections(intersectionFrom, intersectionTo) {
+    if (intersectionFrom === intersectionTo) {
+      return { cost: 0, route: [] };
+    }
+    const sqlVertices = `
```

Comment:
can we extract this part out into a separate method here that returns the bounding box as an object `{ xmin, ymin, xmax, ymax }`, or `null` if one or both vertices are missing?